ClientTests/TabManagerNavDelegateTests.swift

```diff
@@ -0,0 +1,241 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import XCTest
+import WebKit
+
+@testable import Client
+
+class TabManagerNavDelegateTests: XCTestCase {
+    let navigation = WKNavigation()
+
```

Comment:
We could add `sut`, `delegate1` and `delegate2` as variables and use the `setUp` and `tearDown` functions to avoid duplicating the initial setup in each individual test.

Something like:
`override func setUp() {
        super.setUp()
        let sut = TabManagerNavDelegate()
        let delegate1 = WKNavigationDelegateSpy()
        let delegate2 = WKNavigationDelegateSpy()
        }
        
        override func tearDown() {
        super.tearDown()
         let sut = nil
         let delegate1 = nil
         let delegate2 = nil
    }`

Reply:
I've create a helper function `makeSUT`. I think it's cool to keep the test with the given/when/then struct. What do you think?

Reply:
yes it's a nice solution too as long we are avoid repeating code :) and each test gets setup correctly.

What I like about `setUp` and `tearDown` functions is that give you a reliable state because they run before and after each state so reduces flakiness. This [article](https://developer.apple.com/documentation/xctest/xctestcase/set_up_and_tear_down_state_in_your_tests) help to learn more about it

Reply:
I'll check it out. Thanks.